src/Application/View/Extension/DocumentExtension.php

```diff
@@ -62,11 +62,30 @@ public function getFilters(): array
     public function getDocumentIcon(string $filename): string
     {
         /** @var array $CFG_GLPI */
-        global $CFG_GLPI;
+        /** @var \DBmysql $DB */
+        global $CFG_GLPI, $DB;
 
-        $icon = sprintf('/pics/icones/%s-dist.png', strtolower(pathinfo($filename, PATHINFO_EXTENSION)));
+        $extention = strtolower(pathinfo($filename, PATHINFO_EXTENSION));
 
-        return $CFG_GLPI['root_doc'] . (file_exists(GLPI_ROOT . $icon) ? $icon : '/pics/timeline/file.png');
+        $iterator = $DB->request([
```

Comment:
Is a db (on a non indexed field) query really required? According to empty data, icon is just `$ext-dist.png`

Reply:
In my example with a `*.docx` file, `doc**x**-dist.png` doesn't exist, it's `doc-dist.png` (without the X).

Reply:
okay... removing the last letter is probably less huge than a db query anyway :)

Reply:
I'm not sure it'll always be true?

As an intermediate proposal, we can search in DB only if the file without transformation doesn't exist?

Reply:
I guess this can be acceptable this way; yes

Reply:
For ms office documents, you can add extra if file `$ext . 'x'` exists (as far as I know, it's quite common they added a "x" on their historical extensions)
This will prevent DB queries for a type of document that will be quite common I guess